src/presentation/http/policies/userCanUpload.ts

```diff
@@ -0,0 +1,33 @@
+import { isEmpty, notEmpty } from '@infrastructure/utils/empty.js';
+import type { PolicyContext } from '../types/PolicyContext.js';
+import { MemberRole } from '@domain/entities/team.js';
+
+/**
+ * Policy to check whether a user has permission to upload files, if file is a part of note
+ * If note is resolved, we need to check permissions, in other case, we need to check only user authorization
+ *
+ * @param context - Context object, containing Fatify request, Fastify reply and domain services
+ */
+export default async function userCanUploadFileToNote(context: PolicyContext): Promise<void> {
```

Comment:
pls update docs and filename since this policy now check only uploading of the files in Note.

P.S. Do we really need it? Why not to use `userCanEdit` ?

Reply:
How can I use userCanEdit, if request does not include note?

For userCanEdit resolved note is required. And I think, that rewrite its logic for another behaviour is not comfortable way.

Reply:
as i said before `Here i see two cases: to make 2 policies or 2 upload routes (with and without an explicit note)`